test/shared/src/test/scala/monocle/LensInstancesSpec.scala

```diff
@@ -0,0 +1,79 @@
+package monocle
+
+import cats.Eq
+import cats.implicits.catsSyntaxTuple2Semigroupal
+import cats.laws.discipline.{InvariantTests, SemigroupalTests}
+import org.scalacheck.{Arbitrary, Gen}
+import LensInstancesSpec.eqLens
+
+//I'm wondering if this is worth keeping with the
+//sampledEq being deprecated
+class LensInstancesSpec extends MonocleSuite {
+  case class Sample(a: Char, b: Int, c: Boolean)
+  implicit val sampleEq: Eq[Sample] = Eq.fromUniversalEquals
+  implicit val sampleArb: Arbitrary[Sample] = Arbitrary(
+    for {
+      a <- Arbitrary.arbChar.arbitrary
+      b <- Arbitrary.arbInt.arbitrary
+      c <- Arbitrary.arbBool.arbitrary
+    } yield Sample(a, b, c)
+  )
+
+  implicit val arbSampleLensChar: Arbitrary[Lens[Sample, Char]] = Arbitrary(
+    Gen.const(Lens[Sample, Char](_.a)(c => s => s.copy(a = c)))
+  )
+  implicit val arbSampleLensInt: Arbitrary[Lens[Sample, Int]] = Arbitrary(
+    Gen.const(Lens[Sample, Int](_.b)(i => s => s.copy(b = i)))
+  )
+  implicit val arbSampleLensBool: Arbitrary[Lens[Sample, Boolean]] = Arbitrary(
+    Gen.const(Lens[Sample, Boolean](_.c)(b => s => s.copy(c = b)))
+  )
+
+  checkAll("Lens.InvariantLaws", InvariantTests[Lens[Sample, *]].invariant[Char, Int, Boolean])
+
+  checkAll("Lens.SemigroupalLaws", SemigroupalTests[Lens[Sample, *]].semigroupal[Char, Int, Boolean])
+
+  //Explicit tests for clarity
+  test("conflicting lenses are left associative") {
```

Comment:
what do you mean by left associative?

Reply:
maybe wrong language, I meant the right setter operation overwrites the left one, so if they conflict the right most operation (lens replace) result is the result we get

Reply:
I might need to find something better here, the API signature does not point to what's going on
```
zippedLenses.replace(('b', 'c'))
```
Any suggestions?

Reply:
will probably make a better test with fold or foldMap

Reply:
I've cleaned it up, let me know what you think